scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -632,6 +665,36 @@ class ScioContext private[scio] (val options: PipelineOptions,
   }
 
   /**
+    * Get an SCollection for a Pub/Sub topic that includes message attributes.
+    * @group input
+    */
+  def pubsubTopicWithAttributes[T: ClassTag](topic: String,
+                               idLabel: String = null,
+                               timestampLabel: String = null)
+  : SCollection[(T, Map[String, String])] = requireNotClosed {
+    if (this.isTest) {
+      this.getTestInput(PubsubIO(topic))
+    } else {
+      val elementCoder = pipeline.getCoderRegistry.getScalaCoder[T]
+      val outputCoder  = pipeline.getCoderRegistry.getScalaCoder[(T, Map[String, String])]
+      val parseFn = Functions.simpleFn { msg: PubsubMessage =>
```

Comment:
`elementCoder` & `parseFn` are identical in `pubsubSubscriptionWithAttributes` and `pubsubTopicWithAttributes`, put them in a helper method? You can even put almost everything in a helper method and parameterize only the `gio.PubsubIO.read().subscription(topic)` vs `gio.PubsubIO.read().topic(topic)` part.